lib/Star/Kata/Builder/ObjectiveBuilder.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Builder;
+
+use Star\Kata\Model\Objective\ConfigurableObjective;
+use Star\Kata\Model\Objective\Objective;
+
+/**
+ * Class ObjectiveBuilder
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Builder
+ */
+final class ObjectiveBuilder
+{
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var int
+     */
+    private $count = 1;
+
+    public function __construct()
+    {
+        $this->resetData();
+    }
+
+    /**
+     * @return Objective
+     */
+    public function build()
+    {
+        $objective = new ConfigurableObjective($this->data['definition'], $this->data['code']);
```

Comment:
I would like to keep the model clean, but since the builder only use a specific implementation, i feel like the goal of having multiple implementation of objective might not be worth it